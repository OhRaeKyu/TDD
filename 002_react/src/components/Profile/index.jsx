import React from 'react';

export default function index({ userId, userName }) {
  return (
    <div>
      <b>{userId}</b>
      <span>{userName}</span>
    </div>
  );
}
