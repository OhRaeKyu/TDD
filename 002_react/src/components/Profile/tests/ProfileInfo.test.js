import React from 'react';
import { render, screen } from '@testing-library/react';
import Profile from '../index';

describe('ProfileInfo Component', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile userId="oooorx" userName="오래규" />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correctly', () => {
    render(<Profile userId="oooorx" userName="오래규" />);
    screen.getByText('oooorx');
    screen.getByText('오래규');
    screen.getByText(/오/);
  });
});
