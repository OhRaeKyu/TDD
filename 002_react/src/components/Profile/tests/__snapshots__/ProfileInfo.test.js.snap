// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo Component matches snapshot 1`] = `
<div>
  <div>
    <b>
      oooorx
    </b>
    <span>
      오래규
    </span>
  </div>
</div>
`;
